### ISSUE TITLE
Battle Mage sometimes does not throw comet at first turn

### ISSUE THREAD

**Issue by GutZuFusss** [the-marenga/sf-api#116]
First, I hope it is clear which BattleEvents correspond to each log line:
```rust
#### Turn update ####
Attack (from Mage to BattleMage (Attack-Type: 0)
Crit (from Mage to BattleMage)
Damage received (from Mage to BattleMage (dmg: 2)
#### Turn update ####
HP - Left: 18, Right: 8
Attack (from BattleMage to Mage (Attack-Type: 0)
Damage received (from BattleMage to Mage (dmg: 1)
```

The scenario is that the Mage started, and (without looking into it closer yet) it appears that the condition (ref: https://github.com/the-marenga/sf-api/blob/main/src/simulate/mod.rs#L463).
```rust
if attacker.rounds_in_battle == 1 {
```
is not met, since after the first turn of the mage, the following code is executed for the 2nd time before the condition above is checked, if I am not mistaken (ref: https://github.com/the-marenga/sf-api/blob/main/src/simulate/mod.rs#L408-L410).
```rust
        left.rounds_in_battle += 1;
        right.rounds_in_battle += 1;
```

**Comment by the-marenga** [comment]
Yeah, that should probably be either a pre turn thing like guild mushrooms, and/or be put into a classeffect, so it is guaranteed to always trigger and never to trigger twice. I don't have the time rn to fix this, but thanks for pointing that out! I will see when I get around to fixing that
